sklearn/impute/tests/test_impute.py

```diff
@@ -445,6 +445,25 @@ def test_imputation_constant_pandas(dtype):
     assert_array_equal(X_trans, X_true)
 
 
+def test_simple_imputation_missing_value_in_test_array():
+    # [Non Regression Test] Missing value in test set should not throw an error
+    # and return a finite dataset
+    train = [[1], [2]]
+    test = [[3], [np.nan]]
+    imputer = SimpleImputer(add_indicator=True)
+    imputer.fit(train).transform(test)
+
+
+def test_iterative_imputation_missing_value_in_test_array():
```

Comment:
`test_iterative_imputation_missing_value_in_test_array` and `test_simple_imputation_missing_value_in_test_array` can be combined using `pytest`:

```py
@pytest.mark.parametrize('Imputer', (SimpleImputer, IterativeImputer))
def test_imputation_missing_value_in_test_array(Imputer):
    # [Non Regression Test] Missing value in test set should not throw an error
    # and return a finite dataset
    train = [[1], [2]]
    test = [[3], [np.nan]]
    imputer = Imputer(add_indicator=True)
    imputer.fit(train).transform(test)
```

Reply:
Done!